cmd/util/ledger/migrations/storage_v5.go

```diff
@@ -175,6 +183,182 @@ var storageMigrationV5DecMode = func() cbor.DecMode {
 	return decMode
 }()
 
+// Clean-up the payloads by removing broken contracts. This method:
+//   - Removes the contracts code,
+//   - Remove the contract name from account.
+//
+func (m *StorageFormatV5Migration) cleanupBrokenContracts(payloads []ledger.Payload) ([]ledger.Payload, error) {
+	cleanedPayloads := make([]ledger.Payload, 0, len(payloads))
+
+	addToCleanedPayloads := func(payload ledger.Payload) {
+		cleanedPayloads = append(cleanedPayloads, payload)
+	}
+
+	removedNames := make(map[common.AddressLocation]bool, 0)
+	removedContracts := make(map[common.AddressLocation]bool, 0)
+
+	removeBrokenContracts := func(payload ledger.Payload) {
+		keyParts := payload.Key.KeyParts
+
+		rawOwner := keyParts[0].Value
+		rawController := keyParts[1].Value
+		rawKey := keyParts[2].Value
+
+		// If the payload is a cadence value, then do not further process.
+		if !state.IsFVMStateKey(
+			string(rawOwner),
+			string(rawController),
+			string(rawKey),
+		) {
+			addToCleanedPayloads(payload)
+			return
+		}
+
+		address := common.BytesToAddress(rawOwner)
+
+		// Remove the contract name from the account
+		if string(rawKey) == state.KeyContractNames {
+
+			encContractNames := payload.Value
+			if len(encContractNames) == 0 {
+				addToCleanedPayloads(payload)
+				return
+			}
+
+			// Decode contract names
+			contractNames := make([]string, 0)
+			reader := bytes.NewReader(encContractNames)
+			cborDecoder := cbor.NewDecoder(reader)
+			err := cborDecoder.Decode(&contractNames)
+			if err != nil {
+				panic(err)
+			}
+
+			brokenContracts := m.brokenContracts[address]
+
+			// Remove broken contracts
+			updatedContractNames := make([]string, 0)
+			for _, contractName := range contractNames {
+				if !brokenContracts[contractName] {
+					updatedContractNames = append(updatedContractNames, contractName)
+					continue
+				}
+
+				m.Log.Warn().Msgf(
+					"REMOVED contract name '%s' from account: %s",
+					contractName,
+					address,
+				)
```

Comment:
I think it might be safer to dump the contract contents to a file as a backup mechanism

Reply:
Added with a510013 and 68cfb89